aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -629,7 +629,8 @@ static Tensor& logsumexp_out_impl(Tensor& result, const Tensor& self, IntArrayRe
     auto maxes = at::amax(self, dims, true);
     auto maxes_squeezed = (keepdim ? maxes : squeeze_multiple(maxes, dims));
     maxes_squeezed.masked_fill_(maxes_squeezed.abs() == INFINITY, 0);
-    at::sum_out(result, at::exp(self - maxes), dims, keepdim);
+    at::exp_out(result, self - maxes);
```

Comment:
Could this `self - maxes` also be replaced by `at::sub_out(result, self, maxes)`?

Reply:
@vadimkantorov thanks! I didn't know the existence of this function. This would make memory usage `~ 1 x input.shape` which is great!

Reply:
@vadimkantorov once I checked in detail I see that `at::sub_out(result, self, maxes)` does not provide any additional wins over `self-maxes`. I changed exp_out to `.exp_()` to have minimal change.

Unfortunately there is no way to do an all in-place operation unless pytorch has an in-place `logsumexp` function, which it does not.

Thanks for the suggestion anyways!